contrib/move-string-resource.kts

```diff
@@ -0,0 +1,167 @@
+ * all strings.xml files within directories with a prefix of values*
+ * in ../app/src/main/res/ and paste them into
+ * their corresponding files within ../core/src/main/res/values*.
+ *
+ */
+import java.io.File
+import kotlin.system.exitProcess
+
+
+if (args.size < 3) {
+  printCorrectUsageAndExit()
+}
+
+val pathToValues = "/src/main/res/"
+val source = args[0]
+val destination = args[1]
+
+/// name=key
+val key = """name="${args[2]}""""
+
+val sourceDir = File(source + pathToValues)
+if (!sourceDir.exists()) {
+  printModuleIsNotValidDir(source, pathToValues)
+}
+
+println("Running transfer of string resources...")
+val numberOfFilesPastedTo = sourceDir.cutStringResourcesAndPasteToDestination(key, source, destination)
+println("Transfer of string resources complete. Moved $numberOfFilesPastedTo strings.")
+
+fun File.cutStringResourcesAndPasteToDestination(key: String, source: String, destination: String) : Int {
+  var numberOfFilesPastedTo = 0
+  this.walk().filter { it.name.equals("strings.xml") }.forEach { resourceFile ->
+    var cutLine: String = cutLineFromResourceFile(resourceFile, key)
+    if (cutLine.equals("")) {
+      // continue for each loop
+      return@forEach
+    }
+    val destinationFile = openOrCreateDestinationResourceFile(resourceFile, source, destination)
+    pasteLineToDestination(destinationFile, cutLine)
+    numberOfFilesPastedTo++
+  }
+  return numberOfFilesPastedTo
+}
+
+fun openOrCreateDestinationResourceFile(resourceFile: File, source: String, destination: String): File {
+
+  val destinationFilePath = resourceFile.path.replaceRange(0..source.length, destination + "/")
+  val destinationFile = File(destinationFilePath)
+  val destinationDirectoryPath = destinationFilePath
+    .removeRange(
+      (destinationFilePath.length - "strings.xml".length - 1)..destinationFilePath.length - 1
+    )
+  val destinationDirectory = File(destinationDirectoryPath)
+
+  if (!destinationDirectory.exists()) {
+    createDestinationDirectoryAndFile(destinationDirectory, destinationFile)
+  }
+  return destinationFile
+}
+
+fun createDestinationDirectoryAndFile(destinationDirectory: File, destinationFile: File) {
+  destinationDirectory.mkdir()
+  val isNewFileCreated: Boolean = destinationFile.createNewFile()
+  if (!isNewFileCreated) {
+    System.err.println("Could not create resource file ${destinationFile.path}")
+    System.exit(-1)
+  }
+  val initialXMLResourceFileContent = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
+    "<resources xmlns:tools=\"http://schemas.android.com/tools\" tools:ignore=\"DuplicateStrings\">\n" +
+    "</resources>"
+  destinationFile.writeText(initialXMLResourceFileContent)
+  println("Created directory ${destinationDirectory.path} and resource file ${destinationFile.path}")
+}
+
+fun pasteLineToDestination(destinationFile: File, cutLine: String) {
+  var resourceDataWithPastedLine = ""
+  var firstLineRead = false
```

Comment:
I don't think you need this.
to me the flow is 
```
read line
add line to stringbuilder
if (line was "<resources>") add cutLine
```
I might be reading it wrong but it looks like `Add new line` just starts the file off with a `/n`

Reply:
I don't think `read line` keeps newlines, my solution was the firstLineRead bool and to not put newlines if it is false. Couldn't think of a cleaner solution, but if your suggestion works (read line keeps line breaks) I think it is better. 

``` 
fun addNewLine(firstLineRead: Boolean, stringToAddLineTo: String, line: String): String {
...
  if (firstLineRead) {
    tempString = tempString + "\n"
  }
...
```

Reply:
StringBuilder has `appendLn` functions, I think that is what you'll want. My memory is failing me if readLine preserves line endings

Reply:
ok nice! That will look better.